tsfm_public/toolkit/visualization.py

```diff
@@ -194,3 +197,70 @@ def plot_ts_forecasting(
         plt.xticks(rotation=45)
         plt.close()
         return fig
+
+
+def plot_predictions(
+    model: torch.nn.Module,
+    dset: torch.utils.data.Dataset,
+    plot_dir: str = None,
+    num_plots: int = 10,
+    plot_prefix: str = "valid",
+    channel: int = -1,
+    truncate_history: bool = True,
+):
+    """Utility for plotting forecasts along with history.
+
+    Args:
+        model (torch.nn.Module): A trained model.
+        dset (torch.utils.data.Dataset): Dataset that was fed into Trainer for predicting
+        plot_dir (str, optional): A location to save the plot. If None, no plot is saved. Defaults to None.
+        num_plots (int, optional): Number of sub-plots (context windows) to include. Defaults to 10.
+        plot_prefix (str, optional): A prefix for the saved filename. Defaults to "valid".
+        channel (int, optional): Which channels to plot for a multivariate dataset. Defaults to -1.
+        truncate_history (bool, optional): If True only some hisotry (2 * pred_len samples) will be included in the plot. Defaults to True.
+    """
+    # device = torch.cuda.current_device() if torch.cuda.is_available() else torch.device("cpu")
+    device = model.device
+    random_indices = np.random.choice(len(dset), size=num_plots, replace=False)
+    random_samples = torch.stack([dset[i]["past_values"] for i in random_indices]).to(device=device)
+
+    output = model(random_samples)
+    y_hat = output.prediction_outputs[:, :, channel].detach().cpu().numpy()
+    pred_len = y_hat.shape[1]
+
+    # Set a more beautiful style
+    plt.style.use("seaborn-v0_8-whitegrid")
+
+    # Adjust figure size and subplot spacing
+    fig, axs = plt.subplots(num_plots, 1, figsize=(10, 2*num_plots))
```

Comment:
Added an automatic figure size selection, otherwise the aspect ratio changes when different number of figures are required.

Reply:
Thanks